frontend/public/components/dashboard/grid.tsx

```diff
@@ -5,31 +5,48 @@ import { useRefWidth } from '../utils';
 
 export const MEDIA_QUERY_LG = 992;
 
-export const DashboardGrid: React.FC<DashboardGridProps> = ({ mainCards, leftCards, rightCards }) => {
+const mapCardsToGrid = (cards: React.ReactNode[], keyPrefix: string): React.ReactNode[] =>
```

Comment:
#1742 also seems to do such changes. I'd merge that one before this one.

(`React.ComponentType` is more specific than `React.ReactNode`.)